homeassistant/components/switcher_kis/switch.py

```diff
@@ -154,3 +174,89 @@ async def async_turn_on_with_timer_service(self, timer_minutes: int) -> None:
         await self._async_call_api(API_CONTROL_DEVICE, Command.ON, timer_minutes)
         self.control_result = True
         self.async_write_ha_state()
+
+
+class SwitchereShutterChildLockBaseSwitchEntity(SwitcherEntity, SwitchEntity):
+    """Representation of a Switcher shutter base switch entity."""
+
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_icon = "mdi:lock-open"
+    _cover_id: int
+
+    def __init__(self, coordinator: SwitcherDataUpdateCoordinator) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self.control_result: bool | None = None
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """When device updates, clear control result that overrides state."""
+        self.control_result = None
+        self.async_write_ha_state()
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        if self.control_result is not None:
+            return self.control_result
+
+        data = cast(SwitcherShutter, self.coordinator.data)
+        return bool(data.child_lock[self._cover_id] == ShutterChildLock.ON)
```

Comment:
Why don't we always get the state from the coordinator?

Reply:
Background can be found here https://github.com/home-assistant/architecture/discussions/740#discussioncomment-2243081, comment is hidden for no reason but you can still read it.

Switcher devices uses asynchronous UDP broadcasts to update the device state and synchronous TCP API control, for switches this may cause an annoying behavior.
We send a command using TCP to the device, the command succeeded but a UDP update may be already on the way with the old state and the switch will flip-flop.